test/infrastructure/docker/exp/internal/controllers/dockermachinepool_controller.go

```diff
@@ -166,7 +182,39 @@ func (r *DockerMachinePoolReconciler) SetupWithManager(ctx context.Context, mgr
 }
 
 func (r *DockerMachinePoolReconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Cluster, machinePool *expv1.MachinePool, dockerMachinePool *infraexpv1.DockerMachinePool) (ctrl.Result, error) {
-	pool, err := docker.NewNodePool(ctx, r.Client, cluster, machinePool, dockerMachinePool)
+	log := ctrl.LoggerFrom(ctx)
+
+	dockerMachineList, err := getDockerMachines(ctx, r.Client, *dockerMachinePool)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	nodePoolInstances := make([]docker.NodePoolInstance, len(dockerMachineList.Items))
+	for i, dockerMachine := range dockerMachineList.Items {
+		machine, err := util.GetOwnerMachine(ctx, r.Client, dockerMachine.ObjectMeta)
+		if err != nil {
+			log.V(2).Error(err, "failed to get owner machine, skipping")
+			continue
+		}
+		if machine == nil {
+			log.V(2).Info("owner machine not found, skipping", "docker-machine", dockerMachine.Name)
+			continue
+		}
```

Comment:
In this case I expect the DockerMachine owner to be a DockerMachinePool in between the time it's created and the time its ownerMachine is created. In that case we want to avoid prematurely deleting the DockerMachine. I can add an explicit check for this case and delete otherwise.

Reply:
Scratch that. So in this code the DockerMachines have a 1:1 correspondence with the node pool instances and we're using the DockerMachines to maintain an inventory. The code you're referring to is meant to prioritize deleting node pool instances associated with an owner Machine that has the delete machine annotation. So in this case, if we can't find the machine, we can't find the delete annotation so we just don't prioritize deletion of that instance.